File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    username: str | None = None
    action: str
    target_peer_id: int | None
    peer_name: str | None = None
    before_json: dict[str, Any] | None
    after_json: dict[str, Any] | None
    ip_address: str | None
    timestamp: datetime

File: frontend/src/types/audit.ts
export interface AuditLogEntry {
  id: number;
  user_id: number | null;
  username: string | null;
  action: string;
  target_peer_id: number | null;
  peer_name: string | null;
  before_json: Record<string, unknown> | null;
  after_json: Record<string, unknown> | null;
  ip_address: string | null;
  timestamp: string;
}
